lib/Runtime/Math/JavascriptMath.cpp

```diff
@@ -807,6 +818,18 @@ using namespace Js;
         Var JavascriptMath::Subtract_Full(Var aLeft, Var aRight, ScriptContext* scriptContext)
         {
             JIT_HELPER_REENTRANT_HEADER(Op_Subtract_Full);
+            Js::TypeId typeLeft = JavascriptOperators::GetTypeId(aLeft);
```

Comment:
We're supposed to call ToNumeric on each input before checking their types.

Reply:
but ToNumber return a double?

Reply:
Not ToNumber, ToNumeric is a new method defined by the BigInt spec.

Reply:
are you suggesting rewriting the whole subtract flow? or just call toPrimitive?

Technically, if we want to follow strictly to spec then we need to rewrite the whole addition flow too.

Reply:
Calling ToPrimitive is plenty, this looks good to me.

---
In reply to: [228003513](https://github.com/Microsoft/ChakraCore/pull/5792#discussion_r228003513) [](ancestors = 228003513)